File: sdk/samples/all in on code/Visual Studio 2008/CSCustomIEContextMenu/NativeMethods/COMRECT.cs
/****************************** Module Header ******************************\
* Module Name:  COMRECT.cs
* Project:      CSCustomIEContextMenu
* Copyright (c) Microsoft Corporation.
* 
* The class COMRECT is used to define the outer rectangle of the border.
* 
* 
* This source is subject to the Microsoft Public License.
* See http://www.microsoft.com/opensource/licenses.mspx#Ms-PL.
* All other rights reserved.
* 
* THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
* EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
* WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
\***************************************************************************/

using System.Drawing;
using System.Runtime.InteropServices;

namespace CSCustomIEContextMenu.NativeMethods
{
    [StructLayout(LayoutKind.Sequential)]
    public class COMRECT
    {
        public int left;
        public int top;
        public int right;
        public int bottom;
        public override string ToString()
        {
            return string.Concat(new object[] {
                " Left = ", this.left, 
                " Top = ", this.top, 
                " Right = ", this.right,
                " Bottom = ", this.bottom });
        }

        public COMRECT()
        {
        }

        public COMRECT(Rectangle r)
        {
            this.left = r.X;
            this.top = r.Y;
            this.right = r.Right;
            this.bottom = r.Bottom;
        }

        public COMRECT(int left, int top, int right, int bottom)
        {
            this.left = left;
            this.top = top;
            this.right = right;
            this.bottom = bottom;
        }

        public static NativeMethods.COMRECT FromXYWH(int x, int y, int width, int height)
        {
            return new NativeMethods.COMRECT(x, y, x + width, y + height);
        }
    }


}

File: sdk/samples/all in on code/Visual Studio 2008/VBCustomIEContextMenu/NativeMethods/COMRECT.vb
'*************************** Module Header ******************************'
' Module Name:  COMRECT.vb
' Project:      VBCustomIEContextMenu
' Copyright (c) Microsoft Corporation.
' 
' The class COMRECT is used to define the outer rectangle of the border.
' 
' 
' This source is subject to the Microsoft Public License.
' See http://www.microsoft.com/opensource/licenses.mspx#Ms-PL.
' All other rights reserved.
' 
' THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
' EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
' WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
'*************************************************************************'

Imports System.Drawing
Imports System.Runtime.InteropServices

Namespace NativeMethods
    <StructLayout(LayoutKind.Sequential)> _
    Public Class COMRECT
        Public left As Integer
        Public top As Integer
        Public right As Integer
        Public bottom As Integer

        Public Overrides Function ToString() As String
            Return String.Concat( _
                New Object() {" Left = ", Me.left, _
                              " Top = ", Me.top, _
                              " Right = ", Me.right, _
                              " Bottom = ", Me.bottom})
        End Function

        Public Sub New()
        End Sub

        Public Sub New(ByVal r As Rectangle)
            Me.left = r.X
            Me.top = r.Y
            Me.right = r.Right
            Me.bottom = r.Bottom
        End Sub

        Public Sub New(ByVal left As Integer, _
                       ByVal top As Integer, _
                       ByVal right As Integer, _
                       ByVal bottom As Integer)
            Me.left = left
            Me.top = top
            Me.right = right
            Me.bottom = bottom
        End Sub

        Public Shared Function FromXYWH(ByVal x As Integer, _
                                        ByVal y As Integer, _
                                        ByVal width As Integer, _
                                        ByVal height As Integer) _
                                    As NativeMethods.COMRECT
            Return New NativeMethods.COMRECT(x, y, x + width, y + height)
        End Function
    End Class
End Namespace
